samples/durable-task-sdks/dotnet/Agents/PromptChaining/Worker/Orchestrations/ContentCreationOrchestration.cs

```diff
@@ -0,0 +1,67 @@
+using AgentChainingSample.Activities;
+using AgentChainingSample.Shared.Models;
+using Microsoft.DurableTask;
+using Microsoft.Extensions.Logging;
+
+namespace AgentChainingSample.Orchestrations;
+
+/// <summary>
+/// Orchestrates the news article generation workflow with multiple specialized agents
+/// </summary>
+public class ContentCreationOrchestration
+{
+    private readonly ILogger<ContentCreationOrchestration> _logger;
+
+    public ContentCreationOrchestration(ILogger<ContentCreationOrchestration> logger)
+    {
+        _logger = logger;
+    }
+
+    public async Task<ContentWorkflowResult> RunAsync(TaskOrchestrationContext context, ContentCreationRequest request)
+    {
+        _logger.LogInformation("Starting news article generation workflow for topic: {Topic}", request.Topic);
+
+        // 1. Research the topic with Research Agent using web search
+        ResearchData researchData = await context.CallActivityAsync<ResearchData>(
+            "ResearchTopicActivity", 
```

Comment:
Rather than referencing activities using string literals, it's recommended to use `nameof(...)` to identify the target activity or orchestration. That helps prevent typos. Alternatively, the source generator can be used to create type-safe activity calls. See the AspNetWebApp sample for reference on the latter.

Reply:
This is a bit of a pet peeve of mine; while you're correct, using `nameof()` does mean that "innocent" refactorings such as class name changes means potentially breaking existing orchestrations.  Orchestration and activity names are "soft contracts" in that they're recorded and used outside of the assembly and no compiler/runtime will inform you of a break.  As such, the names (IMO) should be separate from the type (e.g. shared as constant strings between caller/callee).  (This is also why I'd typically insist on using `JsonPropertyNameAttribute` even when the property names happen to match.) Anyway, rant over.